segmenter_test.go

```diff
@@ -72,3 +73,62 @@ func TestPathSegmenterEdgeCases(t *testing.T) {
 		}
 	}
 }
+
+type testDepth struct {
```

Comment:
This type and the `arm` field don't seem to really be needed. Your example dot segmenter could just be a plain function.